allennlp/models/semantic_role_labeller.py

```
@@ -0,0 +1,191 @@
+            which knows how to combine different word representations into a single vector per
+            token in your input.
+        verb_indicator: torch.LongTensor, required.
+            A one-hot/all-zeros ``IndexField`` representation of the position of the verb in the sentence.
+            This should have shape (batch_size, num_tokens) and importantly, can be all zeros, in the case
+            that the sentence has no verbal predicate.
+        tags : torch.LongTensor, optional (default = None)
+            A torch tensor representing the sequence of gold labels.  These can either be integer
+            indexes or one hot arrays of labels, so of shape ``(batch_size, num_tokens)`` or of
+            shape ``(batch_size, num_tokens, num_tags)``.
+
+        Returns
+        -------
+        An output dictionary consisting of:
+        logits : torch.FloatTensor
+            A tensor of shape ``(batch_size, num_tokens, tag_vocab_size)`` representing
+            unnormalised log probabilities of the tag classes.
+        loss : torch.FloatTensor, optional
+            A scalar loss to be optimised.
+
+        """
+        embedded_text_input = self.text_field_embedder(tokens)
+        expanded_verb_indicator = verb_indicator.unsqueeze(-1).float()
+        # Concatenate the verb feature onto the embedded text. This now
+        # has shape (batch_size, sequence_length, embedding_dim + 1).
+        embedded_text_with_verb_indicator = torch.cat([embedded_text_input, expanded_verb_indicator], -1)
+        batch_size = embedded_text_with_verb_indicator.size()[0]
+        encoded_text = self.stacked_encoder(embedded_text_with_verb_indicator)
+
+        logits = self.tag_projection_layer(encoded_text)
+        reshaped_log_probs = logits.view(-1, self.num_classes)
+        class_probabilities = F.softmax(reshaped_log_probs).view([batch_size, -1, self.num_classes])
```

I'd recommend making this `.view([batch_size, num_tokens, self.num_classes])`.  That way it will crash here if there's something wrong, instead of failing silently and getting a weird error later.  You can get `num_tokens` up above where you get `batch_size`.